allennlp/commands/train.py

```diff
@@ -299,12 +299,16 @@ def train_model(params: Params,
 
     logger.info("From dataset instances, %s will be considered for vocabulary creation.",
                 ", ".join(datasets_for_vocab_creation))
-    vocab = Vocabulary.from_params(
-            params.pop("vocabulary", {}),
-            (instance for key, dataset in all_datasets.items()
-             for instance in dataset
-             if key in datasets_for_vocab_creation)
-    )
+
+    if os.path.exists(os.path.join(serialization_dir, "vocabulary")):
```

Comment:
This should be `if recover and os.path...`.

Reply:
Sure. Doing right away.